### PR TITLE
Docs: Handle concurrent pipeline execution in separate containers

### LINKED ISSUE #3798: [BUG]: zenml seems to block parallel pipeline calls
### Contact Details [Optional]

_No response_

### System Information

ZENML_LOCAL_VERSION: 0.83.0
ZENML_SERVER_VERSION: 0.83.0
ZENML_SERVER_DATABASE: sqlite
ZENML_SERVER_DEPLOYMENT_TYPE: other
ZENML_CONFIG_DIR: /Users/Harrison.South/Library/Application Support/zenml
ZENML_LOCAL_STORE_DIR: /Users/Harrison.South/Library/Application Support/zenml/local_stores
ZENML_SERVER_URL: http://0.0.0.0:8082
ZENML_ACTIVE_REPOSITORY_ROOT: None
PYTHON_VERSION: 3.12.11
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'mac', 'mac_version': '15.5'}
ACTIVE_PROJECT: default
ACTIVE_STACK: default
ACTIVE_USER: admin
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: 5a854bd5-d998-4dd3-9d5c-6132e2361148
ANALYTICS_USER_ID: a15bff36-09f9-4ba9-8bd7-03d28f95e04a
ANALYTICS_SERVER_ID: fc2f0c9a-dfc3-4bcd-b51c-8850411d32f5
INTEGRATIONS: ['pigeon', 'kaniko', 'numpy', 'airflow', 'sklearn', 'pillow', 'bitbucket', 'pytorch', 'scipy', 'pandas',
'openai']
PACKAGES: {'brotli': '1.1.0', 'deprecated': '1.2.18', 'gitpython': '3.1.44', 'markupsafe': '3.0.2', 'pymysql': 
'1.1.1', 'pysocks': '1.7.1', 'pyyaml': '6.0.2', 'sqlalchemy-utils': '0.41.2', 'accelerate': '1.6.0', 
'aiohappyeyeballs': '2.6.1', 'aiohttp': '3.12.13', 'aiosignal': '1.3.2', 'aiosqlite': '0.21.0', 'alembic': '1.8.1', 
'annotated-types': '0.7.0', 'anyio': '4.9.0', 'asttokens': '3.0.0', 'asyncpg': '0.30.0', 'attrs': '25.3.0', 
'avis-common': '0.1.0', 'banks': '2.1.3', 'bcrypt': '4.0.1', 'beanie': '1.30.0', 'beautifulsoup4': '4.13.4', 'boto3': 
'1.39.0', 'botocore': '1.39.0', 'browserforge': '1.2.3', 'camoufox': '0.4.11', 'certifi': '2025.6.15', 
'charset-normalizer': '3.4.2', 'click': '8.1.7', 'cloudpickle': '2.2.1', 'colorama': '0.4.6', 'comm': '0.2.2', 
'cssselect': '1.3.0', 'cython': '3.1.2', 'dataclasses-json': '0.6.7', 'decorator': '5.2.1', 'dill': '0.4.0', 
'dirtyjson': '1.0.8', 'distro': '1.9.0', 'dnspython': '2.7.0', 'docker': '7.1.0', 'docling': '2.39.0', 'docling-core':
'2.39.0', 'docling-ibm-models': '3.6.0', 'docling-parse': '4.1.0', 'easyocr': '1.7.2', 'et-xmlfile': '2.0.0', 
'etl-service': '0.1.0', 'executing': '2.2.0', 'fastapi': '0.115.14', 'filelock': '3.18.0', 'filetype': '1.2.0', 
'frozenlist': '1.7.0', 'fsspec': '2025.5.1', 'geoip2': '5.1.0', 'gitdb': '4.0.12', 'greenlet': '3.2.3', 'griffe': 
'1.7.3', 'h11': '0.16.0', 'hf-xet': '1.1.5', 'httpcore': '1.0.9', 'httptools': '0.6.4', 'httpx': '0.28.1', 
'huggingface-hub': '0.33.1', 'idna': '3.10', 'imageio': '2.37.0', 'ipython': '9.4.0', 'ipython-pygments-lexers': 
'1.1.1', 'ipywidgets': '8.1.7', 'jedi': '0.19.2', 'jinja2': '3.1.6', 'jiter': '0.10.0', 'jmespath': '1.0.1', 'joblib':
'1.5.1', 'jsonlines': '3.1.0', 'jsonref': '1.1.0', 'jsonschema': '4.24.0', 'jsonschema-specifications': '2025.4.1', 
'jupyterlab-widgets': '3.0.15', 'language-tags': '1.2.0', 'latex2mathml': '3.78.0', 'lazy-loader': '0.4', 
'lazy-model': '0.2.0', 'llama-index-core': '0.12.42', 'loguru': '0.7.3', 'lxml': '5.4.0', 'mako': '1.3.10', 
'markdown-it-py': '3.0.0', 'marko': '2.1.4', 'marshmallow': '3.26.1', 'matplotlib-inline': '0.1.7', 'maxminddb': 
'2.7.0', 'mdurl': '0.1.2', 'motor': '3.7.1', 'mpire': '2.10.2', 'mpmath': '1.3.0', 'multidict': '6.6.3', 
'multiprocess': '0.70.18', 'mypy-extensions': '1.1.0', 'nest-asyncio': '1.6.0', 'networkx': '3.5', 'ninja': 
'1.11.1.4', 'nltk': '3.9.1', 'numpy': '2.3.1', 'openai': '1.93.0', 'opencv-python-headless': '4.11.0.86', 'openpyxl': 
'3.1.5', 'orjson': '3.10.18', 'packaging': '25.0', 'pandas': '2.3.0', 'parso': '0.8.4', 'passlib': '1.7.4', 'pexpect':
'4.9.0', 'pgvector': '0.4.1', 'pillow': '11.3.0', 'pip': '25.1.1', 'platformdirs': '4.3.8', 'playwright': '1.53.0', 
'pluggy': '1.6.0', 'prompt-toolkit': '3.0.51', 'propcache': '0.3.2', 'psutil': '7.0.0', 'psycopg2-binary': '2.9.10', 
'ptyprocess': '0.7.0', 'pure-eval': '0.2.3', 'pyclipper': '1.3.0.post6', 'pydantic': '2.11.1', 'pydantic-core': 
'2.33.0', 'pydantic-settings': '2.10.1', 'pyee': '13.0.0', 'pygments': '2.19.2', 'pylatexenc': '2.10', 'pymongo': 
'4.13.2', 'pymupdf': '1.26.1', 'pyobjc-core': '11.1', 'pyobjc-framework-cocoa': '11.1', 'pypdfium2': '4.30.1', 
'python-bidi': '0.6.6', 'python-dateutil': '2.9.0.post0', 'python-docx': '1.2.0', 'python-dotenv': '1.1.1', 
'python-pptx': '1.0.2', 'pytz': '2025.2', 'rebrowser-playwright': '1.52.0', 'referencing': '0.36.2', 'regex': 
'2024.11.6', 'requests': '2.32.4', 'requests-file': '2.1.0', 'rich': '14.0.0', 'rpds-py': '0.25.1', 'rtree': '1.4.0', 
'ruff': '0.12.1', 's3transfer': '0.13.0', 'safetensors': '0.5.3', 'scikit-image': '0.25.2', 'scikit-learn': '1.7.0', 
'scipy': '1.16.0', 'scrapling': '0.2.99', 'screeninfo': '0.8.1', 'semchunk': '2.2.2', 'sentence-transformers': 
'5.0.0', 'setuptools': '80.9.0', 'shapely': '2.1.1', 'shellingham': '1.5.4', 'six': '1.17.0', 'smmap': '5.0.2', 
'sniffio': '1.3.1', 'socksio': '1.0.0', 'soupsieve': '2.7', 'sqlalchemy': '2.0.41', 'sqlmodel': '0.0.18', 
'stack-data': '0.6.3', 'starlette': '0.46.2', 'sympy': '1.14.0', 'tabulate': '0.9.0', 'tenacity': '9.1.2', 
'threadpoolctl': '3.6.0', 'tifffile': '2025.6.11', 'tiktoken': '0.9.0', 'tldextract': '5.3.0', 'tokenizers': '0.21.2',
'torch': '2.7.1', 'torchvision': '0.22.1', 'tqdm': '4.67.1', 'traitlets': '5.14.3', 'transformers': '4.53.0', 'typer':
'0.16.0', 'typing-extensions': '4.14.0', 'typing-inspect': '0.9.0', 'typing-inspection': '0.4.1', 'tzdata': '2025.2', 
'ua-parser': '1.0.1', 'ua-parser-builtins': '0.18.0.post1', 'urllib3': '2.5.0', 'uvicorn': '0.35.0', 'uvloop': 
'0.21.0', 'w3lib': '2.3.1', 'watchfiles': '1.1.0', 'wcwidth': '0.2.13', 'websockets': '15.0.1', 'widgetsnbextension': 
'4.0.14', 'wrapt': '1.17.2', 'xlsxwriter': '3.2.5', 'yarl': '1.20.1', 'zenml': '0.83.0', 'zstandard': '0.23.0', 
'autocommand': '2.2.2', 'backports.tarfile': '1.2.0', 'importlib-metadata': '8.0.0', 'inflect': '7.3.1', 
'jaraco.collections': '5.1.0', 'jaraco.context': '5.3.0', 'jaraco.functools': '4.0.1', 'jaraco.text': '3.12.1', 
'more-itertools': '10.3.0', 'tomli': '2.0.1', 'typeguard': '4.3.0', 'wheel': '0.45.1', 'zipp': '3.19.2'}

CURRENT STACK

Name: default
ID: 3a5cd57e-0536-4193-b2fd-47f03672a343

ORCHESTRATOR: default

Name: default
ID: 266ab22a-a211-40d5-a9f1-65396d19c62d
Type: orchestrator
Flavor: local
Configuration: {}

ARTIFACT_STORE: default

Name: default
ID: 6f721305-0557-4e96-88fc-ad2d9c17fada
Type: artifact_store
Flavor: local
Configuration: {}

### What happened?

I've set up a FastAPI endpoint that triggers a pipeline in zenml. The code is
```python
pipeline_output = await asyncio.to_thread(
    etl_pipeline,
    input_uri=input_uri,
    file_id=doc_id,
)
```
Where `etl_pipeline` is our zenml pipeline.

When I trigger this endpoint a pipeline trigger successfully. However, whilst the pipeline is running, if I trigger with another document, it gives me the below error.

`Preventing execution of pipeline 'etl_pipeline'. If this is not intended behavior, make sure to unset the environment variable 'ZENML_PREVENT_PIPELINE_EXECUTION'.`

I've tried unsetting this env variable, I've tried setting it to `False` but it still always gives this log and then stops processing the second pipeline.

### Reproduction steps

1.
2.
3.
...

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
Fixes #3798 

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] **IMPORTANT**: I made sure that my changes are reflected properly in the following resources:
  - [ ] [ZenML Docs](https://docs.zenml.io)
  - [ ] Dashboard: Needs to be communicated to the frontend team.
  - [ ] Templates: Might need adjustments (that are not reflected in the template tests) in case of non-breaking changes and deprecations.
  - [ ] [Projects](https://github.com/zenml-io/zenml-projects): Depending on the version dependencies, different projects might get affected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md [hunk 1]
```diff
@@ -746,6 +746,173 @@ You can extend this API to support additional features:
...
+```
+
+### Building the Pipeline Runner Container
+
+Create a dedicated container image for running pipelines. This should be separate from your API server container:
+
+#### Pipeline Runner Dockerfile
+
+```dockerfile
+FROM python:3.9-slim
```
Comment (strickvl) [inline docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md]: 3.9 soon to die. use something higher

File: docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md [hunk 2]
```diff
@@ -746,6 +746,173 @@ You can extend this API to support additional features:
...
+# Each pipeline execution gets its own container instance
+```
+
+#### Container Orchestration Considerations
+
+For production deployments, consider using:
+
+1. **Kubernetes Jobs**: For better resource management and scaling
+2. **Docker Swarm**: For simpler container orchestration
+3. **Cloud Container Services**: Like AWS ECS, Google Cloud Run, or Azure Container Instances
```
Comment (strickvl) [inline docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md]:
```suggestion
2. **Cloud Container Services**: Like AWS ECS, Google Cloud Run, or Azure Container Instances
```

**Comment by htahir1** [comment]
@strickvl @schustmi i changed it a bit to make it more generic
